provisioner.go

```diff
@@ -369,42 +440,48 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
 			},
 		},
 	}
-	lpvVolumeMounts := []v1.VolumeMount{
-		{
-			Name:      "data",
-			ReadOnly:  false,
-			MountPath: parentDir,
-		},
-		{
-			Name:      "script",
-			ReadOnly:  false,
-			MountPath: "/script",
-		},
-	}
 	lpvTolerations := []v1.Toleration{
 		{
 			Operator: v1.TolerationOpExists,
 		},
 	}
 	helperPod := p.helperPod.DeepCopy()
 
+	scriptMount := addVolumeMount(&helperPod.Spec.Containers[0].VolumeMounts, helperScriptVolName, helperScriptDir)
+	scriptMount.MountPath = helperScriptDir
+	dataMount := addVolumeMount(&helperPod.Spec.Containers[0].VolumeMounts, helperDataVolName, parentDir)
+	parentDir = dataMount.MountPath
+	parentDir = strings.TrimSuffix(parentDir, string(filepath.Separator))
+	volumeDir = strings.TrimSuffix(volumeDir, string(filepath.Separator))
+	if parentDir == "" || volumeDir == "" {
+		// it covers the `/` case
+		return fmt.Errorf("invalid path %v for %v: cannot find parent dir or volume dir", action, o.Path)
+	}
+	volumeDir = filepath.Join(parentDir, volumeDir)
+	// Specify the helper pod's env vars.
+	// Vars with empty values are excluded to support default values within the pod template
+	env := filterEmptyEnvVars(append(o.Env,
+		v1.EnvVar{Name: envVolDir, Value: volumeDir},
+		v1.EnvVar{Name: envVolType, Value: string(o.Mode)},
+		v1.EnvVar{Name: envVolSize, Value: strconv.FormatInt(o.SizeInBytes, 10)},
+		v1.EnvVar{Name: envVolName, Value: o.Name},
+		v1.EnvVar{Name: envPVCName, Value: o.PVCName},
+		v1.EnvVar{Name: envPVCNamespace, Value: o.PVCNamespace}))
+
 	// use different name for helper pods
 	// https://github.com/rancher/local-path-provisioner/issues/154
-	helperPod.Name = (helperPod.Name + "-" + string(action) + "-" + name)
+	helperPod.Name = (helperPod.Name + "-" + string(action) + "-" + o.Name)
 	if len(helperPod.Name) > HelperPodNameMaxLength {
 		helperPod.Name = helperPod.Name[:HelperPodNameMaxLength]
 	}
 	helperPod.Namespace = p.namespace
-	helperPod.Spec.NodeName = node
+	helperPod.Spec.NodeName = o.Node
 	helperPod.Spec.ServiceAccountName = p.serviceAccountName
 	helperPod.Spec.RestartPolicy = v1.RestartPolicyNever
 	helperPod.Spec.Tolerations = append(helperPod.Spec.Tolerations, lpvTolerations...)
 	helperPod.Spec.Volumes = append(helperPod.Spec.Volumes, lpvVolumes...)
-	helperPod.Spec.Containers[0].VolumeMounts = append(helperPod.Spec.Containers[0].VolumeMounts, lpvVolumeMounts...)
-	helperPod.Spec.Containers[0].Command = cmdsForPath
-	helperPod.Spec.Containers[0].Args = []string{"-p", filepath.Join(parentDir, volumeDir),
-		"-s", strconv.FormatInt(sizeInBytes, 10),
-		"-m", volumeMode}
```

Comment:
This is a breaking change!
Reading these options within a setup/teardown script is pretty cumbersome as the previous examples show.
Therefore I propose to use environment variables for these parameters. Also env vars potentially provide better backward compatibility than options (imagine sb builds a binary using a CLI framework: it would fail by default when new/unknown options are added while, when using env vars, this is not a problem).
However we could leave the options here for now in addition to the env vars, deprecate the options somehow and remove them later. Though eventually we can remove them now since the feature is pretty young anyway and not many people may rely on it / not many systems have to be changed.
WDYT?

@nicktming this affects the changes you introduced in #127.